ic-os/components/setupos-scripts/check-network.sh

```diff
@@ -203,19 +203,23 @@ function query_nns_nodes() {
 # Establish run order
 main() {
     log_start "$(basename $0)"
-    read_variables
-    get_network_settings
-    print_network_settings
-
-    if [[ -n ${ipv4_address} && -n ${ipv4_prefix_length} && -n ${ipv4_gateway} ]]; then
-        validate_domain_name
-        setup_ipv4_network
-        ping_ipv4_gateway
-    fi
+    if kernel_cmdline_bool_default_true ic.setupos.check_network; then
+        read_variables
+        get_network_settings
+        print_network_settings
+
+        if [[ -n ${ipv4_address} && -n ${ipv4_prefix_length} && -n ${ipv4_gateway} ]]; then
+            validate_domain_name
+            setup_ipv4_network
+            ping_ipv4_gateway
+        fi
 
-    ping_ipv6_gateway
-    assemble_nns_nodes_list
-    query_nns_nodes
+        ping_ipv6_gateway
+        assemble_nns_nodes_list
+        query_nns_nodes
+    else
+        echo "* Network checks skipped by request via kernel command line"
+    fi
```

Comment:
To me, it would make more sense to have all the parameters be *skip* booleans so then the logic would go like:

if kernel_cmdline_bool ic.setupos.skip_network_checks; then
     echo "* Network checks skipped by request via kernel command line"
else
     read_variables
     ...

It's confusing to me that currently in the mainnet path, ic.setupos.check_network is absent, but kernel_cmdline_bool_default_true defaults to true, and therefore we do check network

If they're all "skip" parameters instead, we don't even need the "default" logic. We just return true if the parameter is set and else return false

Reply:
I don't like that idea because it requires people reading the code to think in terms of double negatives.  Having worked in the Dracut initial RAM disk generator myself, they moved away from instituting new parameters that would behave this way because it made the code very difficult to understand.  Of course, the existing parameters stayed for backwards compat reasons.

Reply:
> It's confusing to me that currently in the mainnet path, ic.setupos.check_network is absent, but kernel_cmdline_bool_default_true defaults to true, and therefore we do check network

Not sure I understand this.

```
    if kernel_cmdline_bool_default_true ic.setupos.check_network; then
        read_variables
        get_network_settings
        print_network_settings

        if [[ -n ${ipv4_address} && -n ${ipv4_prefix_length} && -n ${ipv4_gateway} ]]; then
            validate_domain_name
            setup_ipv4_network
            ping_ipv4_gateway
        fi

        ping_ipv6_gateway
        assemble_nns_nodes_list
        query_nns_nodes
    else
        echo "* Network checks skipped by request via kernel command line"
    fi
```

Is it not clear that we're asking if the parameter is true or is absent, we must check?  In other words, to check **is** the default.

Reply:
Maybe we can name the conditional function better?  Something like `boot_flag_or_true`?

Reply:
@Bownairo WDYT? I'll defer to you

Reply:
It doesn't matter to me much which direction this goes, but to @andrewbattat's point I do think it could be simpler to stick with one "default".

The variables we have now are:
```
check_hardware (default true)
check_network (default true)
check_age (default true)
stop_before_installation (default false)
stop_before_reboot (default false)
```

If we pick `false` as default:
```
skip_hardware_check (default false)
skip_network_check (default false)
skip_age_check (default false)
stop_before_installation (default false)
stop_before_reboot (default false)
```

If we pick `true` as default:
```
check_hardware (default true)
check_network (default true)
check_age (default true)
perform_installation (default true)
reboot_after_installation (default true)
```

Reply:
I picked True as default.  Good catch!  Thanks.  I've pushed changes already.

Reply:
Can we now delete kernel_cmdline_bool_default_false if it's no longer in use?